prowler/providers/m365/services/exchange/exchange_service.py

```diff
@@ -53,6 +56,28 @@ def _get_mailbox_audit_config(self):
             )
         return mailboxes_config
 
+    def _get_transport_rules(self):
+        logger.info("Microsoft365 - Getting transport rules configuration...")
+        transport_rules = []
+        try:
+            rules_data = self.powershell.get_transport_rules()
+            if isinstance(rules_data, dict):
+                rules_data = [rules_data]
+            for rule in rules_data:
+                if rule:
+                    transport_rules.append(
+                        TransportRule(
```

Comment:
This implementation checks `if rule:` inside the loop, but won't be better to validate the input data earlier, adding an early return if `rules_data` is empty?
```suggestion
            rules_data = self.powershell.get_transport_rules()
            if not rules_data:
                return transport_rules
            if isinstance(rules_data, dict):
                rules_data = [rules_data]
            for rule in rules_data:
                transport_rules.append(
                    TransportRule(
```

Reply:
Good idea, but this would not check if the rule inside the `rules_data` exists.

Reply:
To cover both cases I've added @andoniaf suggestion maintaining the `if rule` inside the `rules_data` loop.